### PR TITLE
Transactions done on table

### PR DESCRIPTION
**Pull request by Kingno3l**
These are the data implemented in this branch;

- [ ] Database transactions
- [ ] Modified and deleted data in SQL
- [ ] Complex queries that answer analytical questions

277759241-02785557-d4b1-46a2-81cb-1965512d9245
277759339-f59e99aa-e068-4932-9968-3bc2123a094a
277759348-7a360229-7743-48e1-bc5c-91aeda6a2c59
277759354-da521342-0b1c-4db6-a844-cbcc1b10c1a0
277759356-2552df4c-b569-4da3-97fe-6bf1f9acbae1
277759362-b7792e6c-0765-4bd8-bbe3-7430cffbc06f
277759372-238b4e99-8af4-4e92-b141-a13d68a2b808
277759381-452c0c75-e1aa-49a8-b153-008610dcf04f
277759440-8228c128-db92-468f-a052-c05496217812

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -10,3 +10,45 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE NOT name = 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN  10.4 AND 17.3;
+
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
```
Comment (franclobo) [inline queries.sql:18]: - [ ] At the end of your query please add this line 👇🏽 

`SELECT species from animals;`

It is to verify that the change was undone as required 👇🏽 

> Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and **verify that the species columns went back to the state before the transaction.**

File: queries.sql [hunk 2]
```diff
@@ -10,3 +10,45 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+
+BEGIN;
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+SELECT * FROM animals;
+COMMIT;
```
Comment (franclobo) [inline queries.sql:26]: - [ ] Please 🙏🏽  add this line at the end of your query 👇🏽 

``SELECT species from animals;``

It is to verify that the change persists after COMMIT 👇🏽 

> Verify that changes persist after commit.

File: queries.sql [hunk 3]
```diff
@@ -10,3 +10,45 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+
+BEGIN;
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS null;
+SELECT * FROM animals;
+COMMIT;
```
Comment (franclobo) [inline queries.sql:34]: - [ ] Please 🙏🏽  add this line at the end of your query 👇🏽 

``SELECT species from animals;``

It is to verify that the change persists after COMMIT 👇🏽 

> Verify that changes persist after commit.

File: queries.sql [hunk 4]
```diff
@@ -10,3 +10,45 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
```
Comment (franclobo) [inline queries.sql:40]: - [ ] Kindly add this line at the end of your query 👇🏽 

``SELECT COUNT(*) FROM ANIMALS;``

It is to verify if all records in the animals table still exists as required 👇🏽 

> After the rollback verify if all records in the animals table still exists.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @Kingno3l, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- Correct schema ✔️ 
- Correct data ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 2 above)
Comment (Kingno3l) [reply queries.sql:26]: Thank you for the feedback @franclobo
Will that replace line 24?

File: queries.sql (hunk 3 above)
Comment (Kingno3l) [reply queries.sql:34]: Will that replace line 31?

File: queries.sql (hunk 4 above)
Comment (Kingno3l) [reply queries.sql:40]: Will that replace line 36?

File: queries.sql [hunk 5]
```diff
@@ -10,3 +10,48 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE NOT name = 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN  10.4 AND 17.3;
+
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
```
Comment (OyePriscilla) [inline queries.sql:18]: - [ ] In accordance to the project requirement, we are to verify if the changes persist after "ROLLBACK", to achieve this I think you can use the `select` query to check as suggested earlier. 🙏 
- [optional] Kindly update the screenshot in the PR as stated in the requirement. Thanks 

Untitled

File: queries.sql [hunk 6]
```diff
@@ -10,3 +10,48 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+SELECT * FROM animals;
+COMMIT;
+SELECT species from animals;
+
+BEGIN;
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS null;
+SELECT * FROM animals;
+COMMIT;
```
Comment (OyePriscilla) [inline queries.sql:34]: - [ ]  In accordance to the project requirement, we are to verify if the changes persist after "`SET species = 'pokemon'`", to achieve this I think you can use the select query to check as suggested earlier. 🙏

- [optional] Kindly update the screenshot in the PR as stated in the requirement. Thanks 

Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Kingno3l 👏 👏 👏, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Implement `sql` query ✔️ 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kingno3l** [comment]
Thank you.
I just have this review left for this PR.
I am having challenges with pgadmin 4 on my machine so I switched to shell.

I attached 2 screenshot to this PR which covers what you requested and any other request.
Thank you 🥇

Screenshot 2023-11-03 at 22 06 50

***
A different screenshot for further clarity...

Screenshot 2023-11-03 at 22 06 36

**Review by Johnadibe — APPROVED** [review]
# Status: Approved ✔️
### Hi @Kingno3l👋🏽, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

LeonardoInGatsby

## Highlights
- Implemented the changes requested by the previous reviewer✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**